bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/LaunchConfigurationDelegate.java

```diff
@@ -144,45 +161,89 @@ protected GdbLaunch createGdbLaunch(ILaunchConfiguration configuration, String m
 	}
 
 	@Override
-	protected String getGDBVersion(ILaunchConfiguration config) throws CoreException {
+	protected String getGDBVersion(ILaunchConfiguration config) throws CoreException
+	{
 
 		String gdbClientCommand = Configuration.getGdbClientCommand(config, null);
-		String version = DebugUtils.getGDBVersion(config, gdbClientCommand);
-		if (Activator.getInstance().isDebugging()) {
+		String version = getGDBVersion(gdbClientCommand);
+		if (Activator.getInstance().isDebugging())
+		{
 			System.out.println("openocd.LaunchConfigurationDelegate.getGDBVersion " + version);
 		}
 		return version;
 	}
 
+	private String getGDBVersion(String gdbClientCommand) throws CoreException
+	{
+
+		String[] cmdArray = new String[2];
+		cmdArray[0] = gdbClientCommand;
+		cmdArray[1] = "--version";
+
+		List<String> cmdArgsList = Arrays.asList(cmdArray);
+
+		IStatus status;
+		try
+		{
+			Map<String, String> idfEnvMap = new IDFEnvironmentVariables().getEnvMap();
+			ProcessBuilderFactory processBuilderFactory = new ProcessBuilderFactory();
+			status = processBuilderFactory.runInBackground(cmdArgsList, Path.ROOT, idfEnvMap);
+		}
+		catch (IOException e)
+		{
+			throw new DebugException(new Status(IStatus.ERROR, Activator.PLUGIN_ID, DebugException.REQUEST_FAILED,
+					"Error while launching command: " + StringUtils.join(cmdArray, " "), //$NON-NLS-1$
+					e.getCause()));
+		}
```

Comment:
Why can't we replace this piece of code with something like the below? Maybe I'm missing something here
```
String version = DebugUtils.getGDBVersion(config, gdbClientCommand);

//DLL error code handling here
String gdbVersion = LaunchUtils.getGDBVersionFromText(status.getMessage());
		if (gdbVersion == null || gdbVersion.isEmpty())
		{
          ...
		}
```

Reply:
Hi @kolipakakondal.  In the new commit, I replaced getGDBVersion with getGDBVersion from the DebugUtils class with a little modification for DLL not found error handling. So we will have a timeout job and also a clear error message if we have an Error reading GDB STDOUT instead of just writing an exception to the log. Thanks

Reply:
Hi @sigmaaa Thanks for the changes. I'm just curious why the above approach is not considered, which could avoid the redundant code.
```
String version = DebugUtils.getGDBVersion(config, gdbClientCommand);

//DLL error code handling here
String gdbVersion = LaunchUtils.getGDBVersionFromText(status.getMessage());
		if (gdbVersion == null || gdbVersion.isEmpty())
		{
          ...
		}
		
		```